File: 005_bigint_sum_product/program.nim

import bigints

var num = 0.initBigInt

for i in 1'u64 .. 4_000_000'u64 : # sum 4_000_000 numbers
  num = num + i.initBigInt

echo "SUM(1, 4_000_000) = ", num


num = 1.initBigInt

for i in 1'u8 .. 100'u8 : # product 100 numbers (factorial 100)
  num = num * i.initBigInt

echo "PRODUCT(1, 100) = ", num


# https://github.com/nim-lang/bigints
# bigints@1.0.0

File: 005_bigint_sum_product/program.zig

const std = @import("std");

pub fn main() !void{
  //var num: u64 = 0;
  var num: u600 = 0;
  for(0..4000001) |i|{
    num += i;
  }
  
  std.debug.print("SUM(1, 4_000_000) = {d}\n", .{num});
  
  var num2: u600 = 1;
  for(2..100) |i|{
    num2 *= i;
  }
  
  std.debug.print("PRODUCT(1, 100) = {d}\n", .{num2});
}

